ckanext/recombinant/views.py

```diff
@@ -687,6 +759,15 @@ def _process_upload_file(lc: LocalCKAN,
                     # type_ignore_reason: incomplete typing
                     pgerror = e.error_dict['info']['orig'][0].decode(  # type: ignore
                         'utf-8')
+                elif 'fields' in e.error_dict:
+                    # type_ignore_reason: incomplete typing
+                    pgerror = e.error_dict['fields'][0]  # type: ignore
+                    key = re.search(KEY_ERROR_MATCH, str(pgerror))
+                    if key:
+                        key = key.group(1)
+                    else:
+                        key = _('unknown')
+                    raise KeyError(key)
```

Comment:
If this is the only source for this exception, create a new exception class specifically for this case instead of using KeyError here.